tandemhunter-main/TandemHunter.py

```diff
@@ -0,0 +1,465 @@
+        except TypeError:
+            coverage_files = []
+            for root, dirnames, filenames in os.walk(batch_dir):
+                for filename in fnmatch.filter(
+                        filenames, "*{0}".format(cov_file_pattern)):
+                    coverage_files.append(os.path.join(root, filename))
+        except Exception as e:
+            coverage_files = []
+        return coverage_files
+
+    @classmethod
+    def _import_coverage(cls, file, column_dtypes, sep="\t"):
+        """ Method to read output file from coverage file into a dataframe.
+        :param: file: Path to coverage file
+        :param: column_dtypes: Dictionary with column name and datatype for
+                            the given coverage file type
+        :param: sep: seperate used in coverage file (default is tab)
+        :returns: reader iterable containing Pandas dataframe.
+        """
+        reader = pd.read_csv(
+            file,
+            sep=sep,
+            header=0,
+            skipinitialspace=True,
+            na_values="-",
+            dtype=column_dtypes,
+            chunksize=10000,
+        )
+
+        return reader
+
+    @classmethod
+    def _import_intervals(cls, intervals, dup_threshold):
+        """ Method to process interval json file and store as a dictionary
+        :param: file: Path to interval json
+        :param: dup_threshold: Threshold used to identify dup/amp regions
+        :returns: Dictionary
+        """
+        # open json
+        with open(intervals) as json_file:
+            intervals = json.load(json_file)
+
+        # validation json contents
+        for interval in intervals:
+            [interval[k] for k in ["region1", "region2"]]
+            # add on default dup threshold if it has not been defined
+            try:
+                if interval["dup_threshold"] == "":
+                    interval["dup_threshold"] = dup_threshold
+            except KeyError:
+                interval["dup_threshold"] = dup_threshold
+
+        return intervals
+
+    @classmethod
+    def _filter_df(cls, df, filters):
+        """ Filter df.
+        :param: df: Pandas dataframe
+        :parma: filter: Filter values as a dictionary
+        :returns: filtered Pandas dataframe
+        """
+        query = " & ".join(
+            ['{} == "{}"'.format(k, v) for k, v in filters.items()]
+        )
+        filtered_df = df.query(query)
+
+        return filtered_df
+
+    @classmethod
+    def _get_sample_id(cls, file, cov_file_pattern):
+        """ Method to return sample ID from file name.
+        :param: file: Input file name
+        :param: cov_file_pattern: Coverage file suffix to remove from filename
+        :returns: sample id string
+        """
+        sample_id = (
+            os.path.basename(file).split(cov_file_pattern)[0].replace("-", "_")
+        )
+
+        return sample_id
+
+    def _compare_metric_at_intervals(self, cov_chunks, pair, metric):

```

Comment:
Quite a large method: I think this ideally would be split into a renaming/reshaping method, followed by a method which carries out the actual fold-change. It would be easier to write unit tests for if that became a priority in future.